app/assets/javascripts/models/ui/collections_list.js.coffee

```diff
@@ -67,11 +67,20 @@ ns.CollectionsList = do ($ = jQuery
       query = @query.params()
       collections = @collections.results().map (collection) -> collection.id
       index = @collections.results().indexOf(collection)
+
+      exact_match = switch query.free_text
+        when collection.dataset_id then true
+        when collection.id then true
+        when collection.short_name() then true
+        when "#{collection.short_name()}.#{collection.version_id}" then true
```

Comment:
I don't know how detail this ticket asks.
AE_Land.002 is more likely to be type in than AE_Land.2. So maybe a regex match here?